### PR TITLE
[plat-180] Inbound connections

### PR DESCRIPTION
**Pull request by raoulstrackx**
Enables the enclave to listen for incoming connections
This PR requires an enclave-runner based on [rust-sgx PR348](https://github.com/fortanix/rust-sgx/pull/348)

### CODE REVIEW

File: library/std/src/sys/unix/fortanixvme/net.rs [hunk 1]
```diff
@@ -114,6 +117,59 @@ macro_rules! not_available {
...
+            None
+        })
+}
+
+#[derive(Clone, Debug)]
+struct IncomingInfo {
+    peer: Addr,
+    runner_port: u32,
+}
+static INCOMING_INFO: SyncOnceCell<Arc<Mutex<Vec<IncomingInfo>>>> = SyncOnceCell::new();
```
Comment (Pagten) [inline library/std/src/sys/unix/fortanixvme/net.rs:148]: * Is the `Arc` needed here?
* I suppose a `HashMap` would not be more efficient here than a `Vec` because the number of elements will be really small?

File: library/std/src/sys/unix/fortanixvme/net.rs [hunk 2]
```diff
@@ -381,7 +487,47 @@ impl TcpListener {
...
+         * +----[ ]-----+            +-------------+
+         * |   Runner   |            |   enclave   | (2) store peer/runner_port mapping
+         * +--[ ]--[ ]--+            +-[ ]----[ ]--+
+         *     \    \-----  enclave ----/      / (3) Accept new incoming connection from runner
+         *      \-------- proxy --------------/
+         */
+        // (1) Tell the runner to accept an incoming connection on a specific socket.
+        let fd_info = get_fd_info(self.inner.as_raw_fd())
+            .ok_or(io::Error::new(ErrorKind::Other, "Internal error"))?;
+        let mut runner = Client::new(fortanix_vme_abi::SERVER_PORT)?;
```
Comment (Pagten) [inline library/std/src/sys/unix/fortanixvme/net.rs:507]: It's a bit unfortunate that we have to create a new client (and a corresponding new connection to the runner) each time `accept()` is called. This could be a performance improvement for another PR.

**Review by Pagten — APPROVED** [review]

File: library/std/src/sys/unix/fortanixvme/net.rs (hunk 1 above)
Comment (raoulstrackx) [reply library/std/src/sys/unix/fortanixvme/net.rs:148]: - No the `Arc` isn't necessary
- Indeed a `HashMap` is a bit overkill here. In the general case there will only be a single entry in the list

File: library/std/src/sys/unix/fortanixvme/net.rs (hunk 2 above)
Comment (raoulstrackx) [reply library/std/src/sys/unix/fortanixvme/net.rs:507]: Yes there's a separate ticket for this within Fortanix' issue-tracking software.
It'd also be better to let the parent VM contact the enclave instead of the other way around. Now we have to make sure that vsock port `fortanix_vme_abi::SERVER_PORT` isn't used by another application in the parent VM. Reserving a port in the enclave VM isn't an issue as we know exactly what runs there. There's also a ticket for that.

File: library/std/Cargo.toml [hunk 3]
```diff
@@ -42,7 +42,7 @@ dlmalloc = { version = "0.2.1", features = ['rustc-dep-of-std'] }
 fortanix-sgx-abi = { version = "0.3.2", features = ['rustc-dep-of-std'] }
 
 [target.x86_64-unknown-linux-fortanixvme.dependencies]
-fortanix-vme-abi = { git = "https://github.com/fortanix/rust-sgx.git", branch = "master", default-features = false, features = ['rustc-dep-of-std'] }
+fortanix-vme-abi = { git = "https://github.com/fortanix/rust-sgx.git", branch = "raoul/PLAT-182-inbound_connections", default-features = false, features = ['rustc-dep-of-std'] }
```
Comment (raoulstrackx) [inline library/std/Cargo.toml:45]: Need to remove commit [393dd120](https://github.com/fortanix/rust/pull/3/commits/393dd120d1ce17330e156b481d15c574dce706ca) when [rust-sgx PR348](https://github.com/fortanix/rust-sgx/pull/348) is merged.
